packages/panzer/disc-fe/src/Panzer_Evaluator_DomainInterface.hpp

```diff
@@ -0,0 +1,79 @@
+#ifndef PANZER_EVALUATOR_DOMAIN_INTERFACE_HPP
+#define PANZER_EVALUATOR_DOMAIN_INTERFACE_HPP
+
+namespace panzer {
+
+  class Workset;
+
+  /** \brief Mix-in interface to support cell "domains" in panzer.
+   * 
+   *  This class adds support for cell domains into evaluators. This is commonly used in DG discretizations.
+   */
+  class DomainEvaluator {
+
+  public:
+
+    /// Domain types supported by worksets. 
+    enum : int {
+        OWNED=0,    /// All Owned cells for the workset on the MPI process
+        GHOST=1,    /// All Ghosted cells for the workset on the MPI process
+        REAL=2,     /// All Owned and Ghosted cells for the workset on the MPI process
+        VIRTUAL=3,  /// All virtual cells for the workset on the MPI process
+        ALL=4       /// All OWNED, GHOSTED and VIRTUAL cells for the workset on the MPI process
+        };
+
+    /**
+     * \brief Constructor
+     *
+     * \param[in] domain (optional) Cell domain to iterate over
+     */
+    DomainEvaluator(int domain=ALL);
```

Comment:
I think it would be good to have a default constructor that settles on "ALL". This way the "average" panzer evaluator can inherit and then we can end up with the right behavior by default.

Reply:
Agreed. Will add this.